### PR TITLE
Upgrade irqbalance to 1.9.5 and add patches to fix ENOSPC handling

### PR DESCRIPTION
**Pull request by suresh-thelkar**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- Upgraded irqbalance from 1.9.3 → 1.9.5 and updated the source tarball hash in SPECS/irqbalance/irqbalance.signatures.json.
- Dropped the now-unneeded IRQBALANCE_ARGS env patch since it’s covered upstream.
- Added an upstream backport patch to improve behavior under interrupt vector exhaustion (ENOSPC): slot-aware placement (penalize low “slots_left” CPUs) and a fallback mechanism that retries within the configured balance_level scope instead of blacklisting/giving up immediately. These are the changes that I have already contributed to upstream.
- Packaging cleanup: remove unintended %{_prefix}/etc artifacts from the staged install root to avoid shipping config under /usr/etc (SPECS/irqbalance/irqbalance.spec).

###### Change Log  

- NA
###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- NA

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=1097408&view=results

### CODE REVIEW

**Comment by suresh-thelkar** [comment]
@microsoft/cbl-mariner-stable-maintainers, Could you please help review this PR? We are planning to include it in the upcoming monthly release, hence requesting a priority review.

**Comment by kgodara912** [comment]
[Buddy build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=1103250&view=results) after recent force push.

**Comment by suresh-thelkar** [comment]
Marking this PR as a draft for now to test an edge case identified which does not come in normal scenarios.

**Comment by suresh-thelkar** [comment]
Buddy build results after the recent push to GitHub are given below. 
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=1104229&view=results

**Comment by suresh-thelkar** [comment]
Note on PR Checks: One of the ptests for libguestfs is failing, which is a known issue and not related to irqbalance.

**Review by kgodara912 — COMMENTED** [review]
Patch matched with both the upstream merged PRs of `irqbalancer`. The analysis of patch with AI says that the patch is good. The extra changes are few corner cases in original patch and are accepted at upstream quickly. Buddy build is successful.

**Comment by suresh-thelkar** [comment]
> Patch matched with both the upstream merged PRs of `irqbalancer`. The analysis of patch with AI says that the patch is good. The extra changes are few corner cases in original patch and are accepted at upstream quickly. Buddy build is successful.

thanks for the review and sign off.

File: SPECS/irqbalance/irqbalance.spec [hunk 1]
```diff
@@ -33,6 +33,7 @@ make %{?_smp_mflags}
 
 %install
 make DESTDIR=%{buildroot} install
+rm -rf %{buildroot}%{_prefix}/etc
```
Comment (christopherco) [inline SPECS/irqbalance/irqbalance.spec]: question(blocking):  Could you double-check the generated `irqbalance.service` in the built RPM? In 1.9.5, upstream’s service template uses `EnvironmentFile=@pkgconfdir@/irqbalance.env`, and `pkgconfdir` appears to default to `$prefix/etc/default` (`/usr/etc/default` with `--prefix=/usr`). This 
PR then removes `%{buildroot}%{_prefix}/etc` and installs the env file as `/etc/sysconfig/irqbalance`.
 
 I’m worried the final unit may require `/usr/etc/default/irqbalance.env` while the RPM only ships `/etc/sysconfig/irqbalance`, which could make the service fail to start. The old sed for `/path/to/irqbalance.env` may no longer match 1.9.5.

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/irqbalance/irqbalance.spec (hunk 1 above)
Comment (suresh-thelkar) [reply SPECS/irqbalance/irqbalance.spec]: > question(blocking): Could you double-check the generated `irqbalance.service` in the built RPM? In 1.9.5, upstream’s service template uses `EnvironmentFile=@pkgconfdir@/irqbalance.env`, and `pkgconfdir` appears to default to `$prefix/etc/default` (`/usr/etc/default` with `--prefix=/usr`). This PR then removes `%{buildroot}%{_prefix}/etc` and installs the env file as `/etc/sysconfig/irqbalance`.
> 
> I’m worried the final unit may require `/usr/etc/default/irqbalance.env` while the RPM only ships `/etc/sysconfig/irqbalance`, which could make the service fail to start. The old sed for `/path/to/irqbalance.env` may no longer match 1.9.5.

Thanks for the review and catching this. Modified the spec file to fix this packaging issue at the configure step instead of post-install:

- Pass `--with-pkgconfdir=%{_sysconfdir}/sysconfig` and `--sysconfdir=%{_sysconfdir}` so configure substitutes `@pkgconfdir@` → `/etc/sysconfig` directly in the unit. Nothing ever lands under `/usr/etc`, so the `rm -rf %{buildroot}%{_prefix}/etc` is gone.
- Replaced the obsolete `/path/to/irqbalance.env` sed with one that matches the actual 1.9.5 substituted path (`/etc/sysconfig/irqbalance.env` → `/etc/sysconfig/irqbalance`), keeping the historical env file path for upgrade compatibility.
- Install the unit manually (`make install` skips it when pkg-config can't resolve `systemdsystemunitdir`).

Also verified the following after installing the RPM and once again completed all the testing. Everything went well. 
```
sthelkar@TDC4164394045 [ /workspace/AfterUpstrmMerge ]$  rpm -q irqbalance
irqbalance-1.9.5-1.azl3.x86_64
sthelkar@TDC4164394045 [ /workspace/AfterUpstrmMerge ]$ sudo systemctl restart irqbalance
sthelkar@TDC4164394045 [ /workspace/AfterUpstrmMerge ]$ systemctl is-active irqbalance
active
sthelkar@TDC4164394045 [ /workspace/AfterUpstrmMerge ]$ sudo journalctl -u irqbalance -b --no-pager
Apr 30 05:14:54 TDC4164394045 systemd[1]: Started irqbalance.service - irqbalance daemon.
Apr 30 05:31:42 TDC4164394045 systemd[1]: Stopping irqbalance.service - irqbalance daemon...
Apr 30 05:31:42 TDC4164394045 systemd[1]: irqbalance.service: Deactivated successfully.
Apr 30 05:31:42 TDC4164394045 systemd[1]: Stopped irqbalance.service - irqbalance daemon.
Apr 30 05:31:42 TDC4164394045 systemd[1]: Started irqbalance.service - irqbalance daemon.

```

Service starts cleanly, no `EnvironmentFile not found` or non-absolute-path warnings.

Comment (suresh-thelkar) [reply SPECS/irqbalance/irqbalance.spec]: I've once again ran the buddy build. Here are the results - 
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=1106074&view=results

Comment (suresh-thelkar) [reply SPECS/irqbalance/irqbalance.spec]: @christopherco, I've addressed your comments. Can you please take a look at the latest commit and let me know if you have any further feedback.

**Review by christopherco — APPROVED** [review]
:shipit:

**Comment by suresh-thelkar** [comment]
> :shipit:

thank you very much for the review and approval. I will merge this PR shortly.
